Cython/Compiler/ModuleNode.py

```diff
@@ -1592,13 +1593,23 @@ def generate_new_function(self, scope, code, cclass_entry):
             code.putln("new((void*)&(p->%s)) %s();" % (
                 entry.cname, decl_code))
 
+        if scope.internal_mode == 2 and py_attrs:
+            # create a hybrid "optional initialization" where the kwds
+            # argument is used to signal that initialization happens.
+            # This is needed for pickleable closures, where the attributes
```

Comment:
Since this is only "for pickleable closures", there's probably a better way to indicate this specific case than abusing the user controllable `is_internal` option (i.e. `@cython.internal`).

Reply:
I've replaced it with a `needs_pickleable_closure_constructor` flag and reverted `is_internal` back to its original use. Hopefully that's clear, if verbose.

(I think the `is_internal` flag always did more than the user `@cython.internal` option though and is only used for closure scopes)